### DISCUSSION TITLE
Pass PageData to Modal component

### DISCUSSION THREAD

**Discussion by KarolusD** [skeletonlabs/skeleton#1532] [Q&A]
Is there a way to pass PageData to a modal component to handle progressive form enhancement with form actions?

I've tried something like this, but without success:
```
	export let data: PageData

	const modalComponentRegistry: Record<string, ModalComponent> = {
		// Custom Modal 1
		registerModal: {
			// Pass a reference to your custom component
			ref: RegisterModal,
			// Add the component properties as key/value pairs
			props: { data },
			// Provide a template literal for the default component slot
			slot: '<p>Skeleton</p>'
		}
	}

```

Or because Modal component is singleton mounted only once above whole app it's a bad idea to pass PageData there?

**Comment by KarolusD** [comment]
PageData is undefined (not yet accessible) on +layout.svelte hence the problem. This probably mean that it's bad to even try passing PageData to Modal component.

**Reply by Sarenor** [reply]
You can pass any data you want in the ``modalSettings.meta`` field:
https://github.com/skeletonlabs/skeleton/blob/master/src/lib/utilities/Modal/types.ts#L42

https://github.com/skeletonlabs/skeleton/blob/master/src/lib/utilities/Modal/types.ts#L42 -> Accessing Metadata covers how to access that in your custom Modal component
